fendermint/actors/blob_reader/src/state.rs

```diff
@@ -2,30 +2,40 @@
 // Copyright 2021-2023 Protocol Labs
 // SPDX-License-Identifier: Apache-2.0, MIT
 
-use std::collections::HashMap;
-
 use fendermint_actor_blobs_shared::state::Hash;
 use fil_actors_runtime::ActorError;
+use fvm_ipld_blockstore::Blockstore;
 use fvm_ipld_encoding::tuple::*;
-use fvm_shared::{address::Address, MethodNum};
+use fvm_shared::address::Address;
 use log::info;
 
-use crate::shared::{ReadRequest, ReadRequestStatus};
+use crate::shared::{OpenReadRequestTuple, ReadRequest, ReadRequestStatus};
+use recall_ipld::hamt;
+use recall_ipld::hamt::map::TrackedFlushResult;
 
 const MAX_READ_REQUEST_LEN: u32 = 1024 * 1024; // 1MB
 
 /// The state represents all read requests.
-#[derive(Debug, Default, Serialize_tuple, Deserialize_tuple)]
+#[derive(Debug, Serialize_tuple, Deserialize_tuple)]
 pub struct State {
-    /// Map of read requests by request ID.
-    pub read_requests: HashMap<Hash, ReadRequest>,
+    /// ReadRequests Hamt.
+    pub read_requests: ReadRequests,
```

Comment:
Get rid off `HashMap` and uses [`ReadRequests`](https://github.com/recallnet/ipc/pull/537/files#diff-3210cc2f3ca65edf5d3e33cfa6c86687d382908b0cd92c66e3e1d1f2b545205cR153-R176) that is implemented using a `hamt`